### PR TITLE
Fixed issue where PaginatedDataTable would not fill the width of its containing Card

### PR DESCRIPTION
**Pull request by bkonyi**
## Description
Fixed issue where PaginatedDataTable would not fill the width of its containing Card

**Before**
before
**After**
after

## Related Issues

- flutter/flutter#40081

### CODE REVIEW

**Comment by bkonyi** [comment]
The following test is failing, but I'm not exactly sure how to address the failure:

```
00:07 +6: PaginatedDataTable custom horizontal padding - checkbox                                                                                                                                                                                                                          
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════                                         
The following TestFailure object was thrown running a test:                                                                                  
  Expected: <10.0>                      
  Actual: <18.0>  
```

Adjusting the `_customHorizontalMargin` to 18 seems to result in the actual value changing to something larger.

File: packages/flutter/lib/src/material/paginated_data_table.dart [hunk 1]
```diff
@@ -421,73 +421,78 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
...
                   ),
                 ),
               ),
             ),
-          ),
-        ],
+          ],
+        ),
       ),
     );
```
Comment (Hixie) [inline packages/flutter/lib/src/material/paginated_data_table.dart]: you should be able to fix this just by wrapping the Column in a `SizedBox.expand`.

Comment (bkonyi) [reply packages/flutter/lib/src/material/paginated_data_table.dart]: That seems to result in an unconstrained height which the `SingleChildScrollView` doesn't like:

```
════════ Exception caught by rendering library ═════════════════════════════════
The following assertion was thrown during performLayout():
BoxConstraints forces an infinite height.

These invalid constraints were provided to RenderFlex's layout() function by the following function, which probably computed the invalid constraints in question:
  RenderConstrainedBox.performLayout (package:flutter/src/rendering/proxy_box.dart:264:13)
The offending constraints were: BoxConstraints(w=1359.8, h=Infinity)
The relevant error-causing widget was
    PaginatedDataTable 
lib/…/flutter/network_screen.dart:233
When the exception was thrown, this was the stack
#0      BoxConstraints.debugAssertIsValid.<anonymous closure>.throwError 
package:flutter/…/rendering/box.dart:514
#1      BoxConstraints.debugAssertIsValid.<anonymous closure> 
package:flutter/…/rendering/box.dart:558
#2      BoxConstraints.debugAssertIsValid 
package:flutter/…/rendering/box.dart:562
#3      RenderObject.layout 
package:flutter/…/rendering/object.dart:1668
#4      RenderConstrainedBox.performLayout 
...
```

```
════════ Exception caught by rendering library ═════════════════════════════════
RenderBox was not laid out: RenderSemanticsAnnotations#8f706 relayoutBoundary=up1 NEEDS-PAINT
'package:flutter/src/rendering/box.dart':
Failed assertion: line 1691 pos 12: 'hasSize'
The relevant error-causing widget was
    SingleChildScrollView 
```

File: packages/flutter/lib/src/material/paginated_data_table.dart [hunk 2]
```diff
@@ -426,74 +426,79 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
...
-                child: Ink(
-                  height: 64.0,
-                  color: _selectedRowCount > 0 ? themeData.secondaryHeaderColor : null,
-                  child: Padding(
-                    padding: EdgeInsetsDirectional.only(start: startPadding, end: 14.0),
-                    child: Row(
-                      mainAxisAlignment: MainAxisAlignment.end,
-                      children: headerWidgets,
+    return LayoutBuilder(
+        builder: (BuildContext context, BoxConstraints constraints) => Card(
```
Comment (Hixie) [inline packages/flutter/lib/src/material/paginated_data_table.dart]: trivial nit: indentation

File: packages/flutter/lib/src/material/paginated_data_table.dart [hunk 3]
```diff
@@ -426,74 +426,79 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
...
-                    children: footerWidgets,
+            ),
+            DefaultTextStyle(
+              style: footerTextStyle,
+              child: IconTheme.merge(
+                data: const IconThemeData(
+                  opacity: 0.54
+                ),
+                child: Container(
+                  height: 56.0,
```
Comment (Hixie) [inline packages/flutter/lib/src/material/paginated_data_table.dart]: this looks like it won't handle text zoom correctly, fwiw

**Comment by Hixie** [comment]
tests?

File: packages/flutter/lib/src/material/paginated_data_table.dart (hunk 3 above)
Comment (bkonyi) [reply packages/flutter/lib/src/material/paginated_data_table.dart]: Filed https://github.com/flutter/flutter/issues/48522

**Comment by bkonyi** [comment]
Somehow managed to royally screw up the history here... will close and create a new PR.

**Pull request closed without merging**